### ISSUE TITLE
.NET 4.7.1 app referencing fields .NET standard 2.0 can cause missing field exception from select assemblies

### ISSUE THREAD

**Issue by mattscheffer** [dotnet/sdk#1754]
Found in D15Rel 27110.0
1.Create a .NET 4.7.1 C# console app and a ,NET Standard 2.0 class library.
2. Inside the Class Library add the following field/constructor to the class.
```
public System.Net.Http.StringContent StringContent;
        public Class1()
        {
            StringContent = new System.Net.Http.StringContent("asdf");
        }
```
3. Add a reference from the class library to the console app. 
4. Add some code to reference StringContent in the Main method of the console app such as:
```
Class1 c= new Class1();
c.StringContent.GetType();
```
5. Start debugging

Expected:
Application builds fine and runs fine.

Actual:
Application breaks during execution. Following is found in the output:
An unhandled exception of type 'System.MissingFieldException' occurred in Unknown Module.
Field not found: 'ClassLibrary1.Class1.StringContent'.

This also occurs with System.IO.Compression.ZipArchive

**Comment by livarcocc** [comment]
@joperezr can you take a look? Is this one the cases where we created an escape hatch for?

**Comment by livarcocc** [comment]
@mattscheffer just to clarify, in step 3, did you mean add a reference to the class library from the console app?

**Comment by mattscheffer** [comment]
@livarcocc I don't quite remember the bug at the moment, but based on the description I wanted the Console app to be referencing the Class Library.

**Comment by joperezr** [comment]
@livarcocc yeah he meant it the other way around, and also, this problem will be fixed in VS 15.6 Preview 3. What is happening here is that we don't automatically create a binding redirect to System.Net.Http on the apps config, after adding it manually the but doesn't repro anymore:
```xml
    <runtime>
        <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
          <dependentAssembly>
            <assemblyIdentity name="System.Net.Http" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
            <bindingRedirect oldVersion="0.0.0.0-4.2.0.0" newVersion="4.2.0.0" />
          </dependentAssembly>
        </assemblyBinding>
    </runtime>
```

@livarcocc this is the fix that we took in for VS 15.6 preview 3 a few weeks ago by updating the support package that will be inserted.

**Comment by livarcocc** [comment]
@mattscheffer and @joperezr thanks for replying.

Given that we have the fix in already and 15.6 preview3 is out the door, I am going to close this issue.

@mattscheffer if this still reproes with 15.6 Preview3, just comment and we can re-activate and continue investigating.
